limiter_mutexbased.go

```diff
@@ -14,8 +34,8 @@ type mutexLimiter struct {
 	clock      Clock
 }
 
-// New returns a Limiter that will limit to the given RPS.
-func newMutexBased(rate int, opts ...Option) Limiter {
+// newMutexBased returns a new atomic based limiter.
+func newMutexBased(rate int, opts ...Option) *mutexLimiter {
```

Comment:
I actually think we can export these:
- NewMutexBased
- MutexLimiter
- NewAtomicBased
- AtomicLimiter

returning the struct without exporting them isn't really useful. actually I wonder what @prashantv thinks about this?

Reply:
Ops, I responded by email, it didn't attach it here.

Let's copy:

Wait, why?

I don't think we want to expose those structs, and even less the
constructors - we don't intend to expose different implementations, right*?

My understanding was that we're (slowly) moving towards having stable tests
and verified implementations.
Once we have that we will add/benchmark a few more alternatives (mutex
without starvation, ratelimitfx, time/rate), pick the best and stick to
that.

* - we might want to do it if we for whatever reason find that different
implementations work better in some cases, but we're not there yet.

If the concern is breaking changes, we could replace Limiter with a struct
that would wrap/embed the actual implementation.

But I think that's a separate discussion - I'm not changing the API here,
just trying to make the files a bit more organized.

Reply:
ah ok i see your point. agreed that we don't want to expose atmoic/mutex based implementations to the user, which is too deep in the details for callers. it's unfortunate though that the library started out with returning an interface though :(.

Reply:
Meh, we're still in <1.0 - we can change it all ;)

We probably should change it to a struct, but we're not close to 1.0 anyway.